plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.flex.ftl

```diff
@@ -40,9 +40,9 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>javax.servlet.jsp</groupId>
-      <artifactId>javax.servlet.jsp-api</artifactId>
-      <version>2.3.1</version>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>apache-jsp</artifactId>
+      <version>9.4.8.v20171121</version>
       <scope>provided</scope>
```

Comment:
My concern here is that we're now requiring a particular *implementation* version, and thus foisting version maintenance onto the user.

Reply:
We were doing that already. The only question is which implementation version we provide, not whether we provide one.

Reply:
That's not quite true: we were pulling in an API spec jar that also has implementation classes.

With this replacement dependency, the user will be prompted by the `versions-maven-plugin` to update when a new version of Jetty is released.  Should they update?

Reply:
Why do believe the previous version was an API spec jar? As best I can follow the various third party web sites, that is not the case. 

Regardless, CT4E is not requiring a particular implementation version. Both the dev_appserver and presumably the prod runtime are using a  particular implementation version. Becuase this jar is provided, CT4E can't change that. It can only match that as closely as possible.

Reply:
Flex and standard don't seem to be using the same version of Jetty.  Standard uses 9.3. Flex uses 9.4. The version here for standard is 9.3, and for flex 9.4.

I don't believe there is any such BOM or canonical details from App Engine. I think there should be. I hope to convince the App Engine team they need to do a better job of documenting these details. Right now I'm reverse engineering all of this.